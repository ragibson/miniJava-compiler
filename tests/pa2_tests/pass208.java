// PA2 pass qualified ref invocation
class A {
    void p(){
        x.y(3);
    }
}
