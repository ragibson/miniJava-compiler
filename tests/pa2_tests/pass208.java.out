********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "A" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . CallStmt
  .   .   .   QualRef
  .   .   .     "y" Identifier
  .   .   .     IdRef
  .   .   .       "x" Identifier
  .   .   .   ExprList [1]
  .   .   .   . LiteralExpr
  .   .   .   .   "3" IntLiteral
=============================================
